pkg/memory-monitor/src/monitor/util.c

```diff
@@ -143,3 +143,66 @@ int convert_mb_to_bytes_signed(long mb, long *bytes_out) {
     }
     return 0;
 }
+
+char *get_tail(const char *file_path, size_t num_lines) {
+    // Open the file
+    FILE *file = fopen(file_path, "r");
+    if (file == NULL) {
+        syslog(LOG_ERR, "Failed to open file: %s", strerror(errno));
+        return NULL;
+    }
+
+    // Seek to the end of the file
+    fseek(file, 0, SEEK_END);
+    long file_size = ftell(file);
+    if (file_size == -1) {
+        syslog(LOG_ERR, "Failed to get file size: %s", strerror(errno));
+        fclose(file);
+        return NULL;
+    }
+
+    // Start reading the file backwards to find the last `num_lines` lines
+    int lines_read = 0;
+    long i;
+    char c;
+    for (i = file_size - 1; i >= 0; i--) {
+        fseek(file, i, SEEK_SET);
+        c = fgetc(file);
```

Comment:
If we read in chunks and then count the occurrences of \n symbol it'll be tad faster, since we will have less syscalls

Reply:
But it's more complicated as it will require readjusting the seek pointer after finding the proper number of new lines. We can also read more new lines than necessary and handle it. In total, it overcomplicates the code. IMHO, this approach is not that resource-consuming to optimize.

Reply:
ok